src/rpc/invoice.rs

```diff
@@ -324,4 +359,43 @@ where
             )),
         }
     }
+
+    async fn settle_invoice(
+        &self,
+        params: SettleInvoiceParams,
+    ) -> Result<SettleInvoiceResult, ErrorObjectOwned> {
+        let network_actor = self.network_actor.as_ref().ok_or(ErrorObjectOwned::owned(
+            CALL_EXECUTION_FAILED_CODE,
+            "network actor not initialized".to_string(),
+            Option::<()>::None,
+        ))?;
+
+        let SettleInvoiceParams {
+            ref payment_hash,
+            ref payment_preimage,
+        } = params;
+
+        let message = move |rpc_reply| -> NetworkActorMessage {
+            NetworkActorMessage::Command(NetworkActorCommand::SettleInvoice(
+                *payment_hash,
+                *payment_preimage,
+                rpc_reply,
+            ))
+        };
+
+        match call_t!(network_actor, message, RPC_TIMEOUT_MS).map_err(|ractor_error| {
```

Comment:
maybe can reuse macro handle_actor_call like rpc/channel.rs.

Reply:
Forgot about this sweet. Thanks.